lib/ballot.go

```diff
@@ -361,10 +384,10 @@ func (b *BallotBoxes) AddBallot(ballot Ballot) (isNew bool, err error) {
 	}
 
 	// unknown ballot will be in `WaitingBox`
-	if ballot.State() == sebakcommon.BallotStateINIT {
-		err = b.AddVotingResult(vr, b.WaitingBox)
-	} else {
+	if ballot.IsFitInVoting() {
 		err = b.AddVotingResult(vr, b.VotingBox)
+	} else {
+		err = b.AddVotingResult(vr, b.WaitingBox)
 	}
```

Comment:
This block is repeated. I think you can insert this block to `BallotBoxes` like this,
```go
			if ballot.IsFitInVoting() {
				err = b.AddVotingResult(vr, b.VotingBox)
			} else {
				err = b.AddVotingResult(vr, b.WaitingBox)
			}
```
to
```go
func (b *BallotBoxes) AddVotingResult(vr *VotingResult, ballot Ballot) (err error) {
	b.Lock()
	defer b.Unlock()

	b.Results[vr.MessageHash] = vr

	if ballot.IsFitInVoting() {
		err = b.VotingBox.AddVotingResult(vr)
	} else {
		err = b.WaitingBox.AddVotingResult(vr)
	}

	return
}
```